src/fhirpath/engine/es/__init__.py

```diff
@@ -178,7 +178,9 @@ def _execute(self, query, unrestricted, query_type):
             calculate_field_index_name=self.calculate_field_index_name,
             get_mapping=self.get_mapping,
         )
-        if query_type == EngineQueryType.DML:
+        if query._scroll_id:
```

Comment:
To be able to switch on a query type, I think I'd need to use a new method instead of `fetchall` in which I'd pass a new type (this is how `count` works).
What do you think?

Reply:
let's add a query.scroll method and introduce a new query_type